src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ApproverCaseloadService.kt

```diff
@@ -3,24 +3,141 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.caseload
       return emptyList()
     }
-    return licences
+
+    val nomisIds = licences.mapNotNull { it.nomisId }
+    val deliusRecords = probationSearchApiClient.searchForPeopleByNomsNumber(nomisIds)
+    val deliusNomisIds = deliusRecords.mapNotNull { it.otherIds.nomsNumber }
+    val nomisRecords = prisonerSearchApiClient.searchPrisonersByNomisIds(deliusNomisIds)
+
+    val prisonerRecord: List<Triple<PrisonerSearchPrisoner, OffenderDetail?, LicenceSummaryApproverView?>> =
+      nomisRecords.map {
+        Triple(it, it.getDeliusRecord(deliusRecords), it.getLicenceSummary(licences))
+      }
+
+    val caseload = prisonerRecord.map { (nomisRecord, deliusRecord, licenceSummary) ->
+      val releaseDate = when (licenceSummary) {
+        null -> nomisRecord.selectReleaseDate()
+        else -> licenceSummary.actualReleaseDate ?: licenceSummary.conditionalReleaseDate
+      }
+      ApprovalDetails(
+        deliusRecord = deliusRecord,
+        comUsernameOnLicence = licenceSummary?.comUsername,
+        licenceSummary = ApprovalLicenceSummary(
+          licenceId = licenceSummary?.licenceId,
+          name = "${nomisRecord.firstName} ${nomisRecord.lastName}".convertToTitleCase(),
+          prisonerNumber = nomisRecord.prisonerNumber,
+          submittedByFullName = licenceSummary?.submittedByFullName,
+          releaseDate = releaseDate,
+          urgentApproval = licenceSummary?.isDueToBeReleasedInTheNextTwoWorkingDays,
+          approvedBy = licenceSummary?.approvedByName,
+          approvedOn = licenceSummary?.approvedDate,
+          isDueForEarlyRelease = licenceSummary?.isDueForEarlyRelease,
+        ),
+      )
+    }
+
+    val comUsernames = caseload.mapNotNull { it.comUsernameOnLicence }
+    val coms = communityApiClient.getStaffDetailsByUsername(comUsernames)
+
+    val approvalCases = caseload.map {
+      ApprovalCase(
+        licenceId = it.licenceSummary?.licenceId,
+        name = it.licenceSummary?.name,
+        prisonerNumber = it.licenceSummary?.prisonerNumber,
+        submittedByFullName = it.licenceSummary?.submittedByFullName,
+        releaseDate = it.licenceSummary?.releaseDate,
+        urgentApproval = it.licenceSummary?.urgentApproval,
+        approvedBy = it.licenceSummary?.approvedBy,
+        approvedOn = it.licenceSummary?.approvedOn,
+        isDueForEarlyRelease = it.licenceSummary?.isDueForEarlyRelease,
+        probationPractitioner = findProbationPractitioner(it.deliusRecord, it.comUsernameOnLicence, coms),
+      )
+    }.sortedByDescending { it.releaseDate }
+
+    return approvalCases
   }
+}
 
-  companion object {
-    val log: Logger = LoggerFactory.getLogger(this::class.java)
+fun PrisonerSearchPrisoner.getDeliusRecord(deliusRecords: List<OffenderDetail>): OffenderDetail? {
+  return deliusRecords.find { it.otherIds.nomsNumber == this.prisonerNumber }
+}
+
+fun PrisonerSearchPrisoner.getLicenceSummary(licences: List<LicenceSummaryApproverView>): LicenceSummaryApproverView? {
```

Comment:
**quibble(non-blocking)**:  I'd probably be tempted to swap the target of the extension function round: 

`licenceSummaries.findFor(prisoner)`

The reason why is that licenceSummaries isn't really an attribute of prisoner search prisoner:
```kt
val prisoners = prisonSearchApi.getPrisoners()
val licence = prisoner.getLicenceSummary()
```
makes it look like the licences are stored in the prisonSearchApi which is potentially a bit confusing.